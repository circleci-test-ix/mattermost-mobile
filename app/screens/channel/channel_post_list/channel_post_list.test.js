// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Preferences} from 'mattermost-redux/constants';

import ChannelPostList from './channel_post_list';

describe('ChannelPostList', () => {
    const baseProps = {
        actions: {
            loadPostsIfNecessaryWithRetry: jest.fn(),
            loadThreadIfNecessary: jest.fn(),
            increasePostVisibility: jest.fn(),
            selectPost: jest.fn(),
            recordLoadTime: jest.fn(),
            refreshChannelWithRetry: jest.fn(),
        },
        channelId: 'current_channel_id',
        channelRefreshingFailed: false,
        currentUserId: 'current_user_id',
        lastViewedAt: 12345,
        loadMorePostsVisible: false,
        postIds: [],
        postVisibility: 15,
        refreshing: false,
        navigator: {
            pop: jest.fn(),
            setButtons: jest.fn(),
            setOnNavigatorEvent: jest.fn(),
        },
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <ChannelPostList {...baseProps}/>,
        );
        expect(wrapper.getElement()).toMatchSnapshot();

        wrapper.setProps({postIds: null});
        expect(wrapper.getElement()).toMatchSnapshot();

        wrapper.setProps({postIds: undefined});
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
