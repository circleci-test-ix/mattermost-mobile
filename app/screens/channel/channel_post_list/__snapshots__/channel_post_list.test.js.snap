// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPostList should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(PostList)
    channelId="current_channel_id"
    currentUserId="current_user_id"
    extraData={false}
    indicateNewMessages={true}
    lastViewedAt={12345}
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    onLoadMoreUp={[Function]}
    onPostPress={[Function]}
    postIds={Array []}
    refreshing={false}
    renderFooter={[Function]}
    renderReplies={true}
  />
  <Connect(AnnouncementBanner)
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
  />
  <Connect(RetryBarIndicator) />
</View>
`;

exports[`ChannelPostList should match snapshot 2`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(PostList)
    channelId="current_channel_id"
    currentUserId="current_user_id"
    extraData={false}
    indicateNewMessages={true}
    lastViewedAt={12345}
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    onLoadMoreUp={[Function]}
    onPostPress={[Function]}
    postIds={Array []}
    refreshing={false}
    renderFooter={[Function]}
    renderReplies={true}
  />
  <Connect(AnnouncementBanner)
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
  />
  <Connect(RetryBarIndicator) />
</View>
`;

exports[`ChannelPostList should match snapshot 3`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(PostList)
    channelId="current_channel_id"
    currentUserId="current_user_id"
    extraData={false}
    indicateNewMessages={true}
    lastViewedAt={12345}
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
    onLoadMoreUp={[Function]}
    onPostPress={[Function]}
    postIds={Array []}
    refreshing={false}
    renderFooter={[Function]}
    renderReplies={true}
  />
  <Connect(AnnouncementBanner)
    navigator={
      Object {
        "pop": [MockFunction],
        "setButtons": [MockFunction],
        "setOnNavigatorEvent": [MockFunction],
      }
    }
  />
  <Connect(RetryBarIndicator) />
</View>
`;
